web/setups/add.php

```diff
@@ -0,0 +1,45 @@
+<?php
+/**
+ * Copyright (c) 2017 University of Illinois, NCSA.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the
+ * University of Illinois/NCSA Open Source License
+ * which accompanies this distribution, and is available at
+ * http://opensource.ncsa.illinois.edu/license.html
+ */
+
+// This page adds the configuration edited in edit.php to in config.php
+include 'core.php';
+
+// open a new temprory file to write data to it
+$file = fopen('../config.php.temp', "w+") or die('Cannot open file: Check whether file exist and it have correct permissions');
```

Comment:
how about using php tmpfile and copying the result to config.php, this would make it so only config.php needs to be writeable by the webserver.